### PR TITLE
CI: Add checks for linting, building APK and building web

### LINKED ISSUE #196: CI: Adding basic checks in a new workflow
The application right now lacks tests, we can add a couple of basic ones -
- Lint tests (would make sure that no one is using a deprecated API or an older version of `Flutter`)
- Making sure the apk builds (would make sure that a PR is not destroying the app completely)

These tests will run on push and on a PR which might make the reviewing process a bit easier. I can take this up if we plan on adding this!

### PR DESCRIPTION
**Pull request by Saransh-cpp**
## Changes/Additions
- Added lint instructions in `analysis_options.yaml`
- Formatted all the files (to pass the lint check)
- Added a new workflow to check lint, build apk, and build web

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
I think I still had that old commit somewhere in my git history (probably committed to master at that time). Also, weirdly the workflows are not running, I'll look into it.

**Review by yashlamba — CHANGES_REQUESTED** [review]
@Saransh-cpp can you remove all the other changes from this PR and just keep the CI, if you feel other things are relevant, you can open another PR. This will make reviewing a lot easier.

**Comment by Saransh-cpp** [comment]
On it!

**Comment by Saransh-cpp** [comment]
I will create another PR to lint all the files, fixing the lint check.

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Checks
```
Comment (yashlamba) [inline .github/workflows/checks.yml]:
```suggestion
name: build-checks
```

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+on:
+  push:
+  pull_request:
+
+  # everyday at 3 am UTC
+  schedule:
+    - cron:  '0 3 * * *'
+
```
Comment (yashlamba) [inline .github/workflows/checks.yml]:
```suggestion
```

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable'
+      - name: Lint
+        run: |
+          flutter pub get
+          flutter analyze
+
+  build-apk-and-web:
```
Comment (yashlamba) [inline .github/workflows/checks.yml]: Let's separate out the builds for web and android.

**Review by yashlamba — CHANGES_REQUESTED** [review]

**Comment by Saransh-cpp** [comment]
Closing this PR.

I am cleaning up my open PRs and closing the stale ones. Please let me know if the changes are required in the future. I would be more than happy to re-open and work on this.

**Pull request closed without merging**